src/hepattn/models/decoder.py

```diff
@@ -262,18 +268,21 @@ def __init__(
         dense_kwargs = dense_kwargs or {}
 
         residual = partial(Residual, dim=dim)
-        self.q_ca = residual(Attention(dim, qkv_norm=qkv_norm, **attn_kwargs), norm=attn_norm)
-        self.q_sa = residual(Attention(dim, qkv_norm=qkv_norm, **attn_kwargs), norm=attn_norm)
+        self.norm_ca = getattr(nn, attn_norm)(dim) if attn_norm else nn.Identity()
+        self.norm_sa = getattr(nn, attn_norm)(dim) if attn_norm else nn.Identity()
```

Comment:
I don't think we want to pull these additional normalisations out, this could cause double normalisation if `qkv_norm` is already enabled. I would just rely on `qkv_norm` here and remove the extra norms